src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1827,22 +1844,22 @@ export default class AircraftModel {
      * @return {number}
      */
     _calculateTargetedAltitudeVnavDescent(hardRestrictedWaypointModel) {
-        let waypointAltitude = hardRestrictedWaypointModel.altitudeMaximum;
-        const softOrHardRestrictedWaypoint = this.fms.nextAltitudeRestrictedWaypoint;
-        const nextFixIsSoftlyRestricted = softOrHardRestrictedWaypoint.name !== hardRestrictedWaypointModel.name;
-        const nextFixIsAtOrAboveRestriction = softOrHardRestrictedWaypoint.altitudeMinimum !== INVALID_NUMBER;
-
-        // NOTE: Currently does not cover any "AT/BELOW", since we will already be descending to bottom anyway
-        if (nextFixIsSoftlyRestricted && nextFixIsAtOrAboveRestriction) {
-            waypointAltitude = softOrHardRestrictedWaypoint.altitudeMinimum;
-        }
-
         // TODO: This could be improved by making the descent at the exact rate needed to reach
         // the altitude at the same time as reaching the fix. At this point, the problem is that
         // while we DO know the descent rate and descent angle to shoot for, we don't know the
         // length of time before the next update, so we can't accurately estimate the altitude to
         // target in the current iteration.
-        return Math.max(waypointAltitude, this.mcp.altitude);
+
+        const nextRestrictedWaypoint = this.fms.nextAltitudeRestrictedWaypoint;
+        if (_isNil(nextRestrictedWaypoint)) {
+            return this.fms.getBottomAltitude();
+        }
```

Comment:
Probably right here, not sure what the really correct thing is at the moment

Reply:
After thinking about it, yes, i think it is `this.mcp.altitude` we should be looking at here.

Reply:
I believe `this.fms.getBottomAltitude()` is the correct thing here.  It won't return anything useful right now because the FMS doesn't actually plan the bottom altitude so it will do nothing currently.  However in the future the FMS should return the correct value.
`this.mcp.altitude` is likely to hold an incorrect value.  There are basically two times it would be used, and both occur with at least one waypoint at the end of the procedure without an altitude restriction.  The first is an automatic descent from cruise altitude which would currently leave the cruise altitude in the MCP.  The second is if a controller issued an altitude clearance and later issued a descend via star clearance,  the MCP would hold either an invalid number or the altitude clearance.